kasa/cli.py

```diff
@@ -440,6 +440,23 @@ async def print_discovered(dev: SmartDevice):
         credentials=credentials,
     )
 
+    # TimeoutError could have cancelled discover() before all
+    # discovered devices can print
+    not_printed = [
+        device
+        for ip, device in discovered_devices.items()
+        if ip not in discovered and ip not in auth_failed
+    ]
+    for device in not_printed:
+        await print_discovered(device)
+
+    for device in discovered_devices.values():
+        await device.protocol.close()
+        # TODO transports need to permit close to avoid
+        # ERROR:asyncio:Unclosed client session
```

Comment:
I think we will want to fix this properly, so I'll mark this PR as a draft to make it easier to keep track on things that are ready to be merged.